### ISSUE TITLE
rr hangs while recording oss-fuzz test case in docker container

### ISSUE THREAD

**Issue by rxEckT** [mozilla/rr#2357]
I'm trying to replicate the issue in https://github.com/glennrp/libpng/issues/275 with rr.

### Summary
rr successfully records the png_uar.c executable by kcc.  
However, rr hangs when trying to record the fuzzing test case by kcc when in the oss-fuzz docker container.

### Details

libpng revision used: 0a882b57876776ab7c46f13cf170b8d376036a6a  
oss-fuzz issue: https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=12803  
libpng issue: https://github.com/glennrp/libpng/issues/275  
rr version: 5.2.0

```
Host Operating System: Ubuntu 18.10
          Host Kernel: Linux 4.18.0-17-generic
    Host Architecture: x86-64
``` 

```
Host clang:
clang version 7.0.0-3 (tags/RELEASE_700/final)
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
```
 
```
$ docker version
Client:
 Version:           18.09.6
 API version:       1.39
 Go version:        go1.10.8
 Git commit:        481bc77156
 Built:             Sat May  4 02:35:29 2019
 OS/Arch:           linux/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          18.09.6
  API version:      1.39 (minimum version 1.12)
  Go version:       go1.10.8
  Git commit:       481bc77
  Built:            Sat May  4 01:59:36 2019
  OS/Arch:          linux/amd64
  Experimental:     false
```
---
# TL;DR & How to reproduce

Hello, I've run into an issue where rr hangs while recording a libfuzzer binary built from the oss-fuzz project. Details here & How to reproduce: https://gist.github.com/rxEckT/a04d3ccfdfb8b53051d62f69714bfe08

**Comment by rxEckT** [comment]
Could this issue be related to #2329 by any chance?

**Comment by rocallahan** [comment]
Yes, this is exactly #2329.

**Comment by rocallahan** [comment]
Please try this with master. It should be fixed on master.

**Comment by rxEckT** [comment]
I've confirmed that it works on master. Thanks!
